validator_worker/src/follower.rs

```diff
@@ -1,12 +1,139 @@
-use futures::future::FutureExt;
-use primitives::validator::{Validator, ValidatorFuture};
-use primitives::Channel;
+use std::error::Error;
 
-#[derive(Clone)]
-pub struct Follower {}
+use primitives::adapter::Adapter;
+use primitives::validator::{ApproveState, MessageTypes, NewState, RejectState};
+use primitives::BalancesMap;
 
-impl Validator for Follower {
-    fn tick(&self, _channel: Channel) -> ValidatorFuture<()> {
-        futures::future::ok(()).boxed()
+use crate::core::follower_rules::{is_healthy, is_valid_transition};
+use crate::heartbeat::heartbeat;
+use crate::sentry_interface::SentryApi;
+use crate::{get_state_root_hash, producer};
+use chrono::Utc;
+
+enum InvalidNewState {
+    RootHash,
+    Signature,
+    Transition,
+}
+
+enum NewStateResult {
```

Comment:
this is internal `Result` type, as @Ivshti mentioned, things are error because invalid transactions, shouldn't implement Error.
Atm if we use Result, instead of custom Error enum, we will need to handle the Error somehow and I am not sure if this is sufficient and if I understood @Ivshti correct.